docs/man-openmpi/man3/MPI_Type_create_struct.3.rst

```diff
@@ -73,8 +73,115 @@ DESCRIPTION
 :ref:`MPI_Type_create_struct` creates a structured data type. This routine
 replaces :ref:`MPI_Type_struct`, which is now deprecated.
 
-NOTE - This routine replaces :ref:`MPI_Type_struct`, which is deprecated. See
-the man page :ref:`MPI_Type_struct` for information about that routine.
+
+:ref:`MPI_Type_create_struct` is the most general type constructor. It further
+generalizes :ref:`MPI_Type_create_hindexed` in that it allows each block to consist of
+replications of different datatypes.
+
+**Example 1:**
+Let ``type1`` have type map
+
+::
+
+
+       {(double, 0), (char, 8)}
+
+with extent 16. Let ``B = (2, 1, 3)``, ``D = (0, 16, 26)``, and ``T = (MPI_FLOAT, type1, MPI_CHAR)``.
+Then a call to ``MPI_Type_create_struct(3, B, D, T, newtype)``
+returns a datatype with type map
+
+::
+
+
+       {
+        (float, 0), (float,4),             // 2 float
+        (double, 16), (char, 24),          // 1 type1
+        (char, 26), (char, 27), (char, 28) // 3 char
+       }
+
+That is, two copies of ``MPI_FLOAT`` starting at 0, followed by one copy of
+``type1`` starting at 16, followed by three copies of ``MPI_CHAR``, starting at
+26. (We assume that a float occupies 4 bytes.)
```

Comment:
Rather than the parenthetical with `We assume...`, you might want to put the assumptions about sizes that apply to all the examples before Example 1.  Don't use the 1st person; perhaps say something like "The examples below assume a platform with the following type sizes ..."

Reply:
I think I'll just leave it out for the reader. It becomes unnecessarily complicated to my taste.